docs/DeveloperGuide.md

```diff
@@ -4,10 +4,150 @@
+- Prints`>`before the user's input, to help the user differentiate between their input and an output from the
+application.
+- Reads input from the user.
+
+<br/>
+
+### Parser component
+The source code can be found in [Parser.java](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Parser.java)
+
+![parser_diagram](images/parser_diagram.png)
+<br/> Fig 4 - Parser Diagram
+
+The Parser component consists of a`Parser`, `Functions`, `ExpenditureList` and `Expenditure`class
+
+The Parser component:
+- Receives user's input and splits it into the Command Type and Description using the`Functions`class.
+- Uses the`itemList`, which is an`ExpenditureList`object, to instantiate a`Command`object based on the Command Type.
+- Returns the`Command`object that can then be executed.
+
+We pass in the`itemList`to the`Command`object instead of using a global variable to ease testing. This way, we can 
+add, delete and update`Expenditure`entries in a new`itemList`during testing without affecting the actual`itemList`.
+
+### Command component 
+The source can be found in [command](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/command)
+
+![command_diagram](images/command_diagram.png)
+<br/> Fig 5 - Command Diagram
+
+The Command component consists of `Command` abstract class, `AddCommand`, `ByeCommand`, `DeleteCommand`, `HelpCommand`
+, `ListCommand`and `UpdateCommand` that extends the `Command` class.
+
+The Command component:
+- Provides all the Command classes which can be instantiated by `Parser.parseCommand()`. The Command objects can then be 
+executed. Only 1 Command object can be created.
+- Includes a `.executeCommand()` method in each Command classes which performs the relevant command and throws 
+exceptions if an error is encountered. The error is then handled.
+
+### Storage component 
+The source can be found in [Storage.java](https://github.com/AY2122S2-CS2113T-T10-4/tp/blob/master/src/main/java/seedu/mindmymoney/Storage.java)
+
+![storage_diagram](images/storage_diagram.png)
+<br/> Fig 6 - Storage Diagram
+
+The Storage component consists of `Storage` class.
+
+The Storage component:
+- `MMM` class initialises a `Storage` object upon start up. The `Storage` class consists of 
+`load()` and `save()` methods. 
+- Concurrently, `MMM` will call the `load()` method and load any data that is stored on the hard disk.
+- `MMM` calls the `save()` method and stores remaining data onto the hard disk when the program exits.
+
+
+## Implementation
+This section describes some noteworthy details on how certain features are implemented.
+
+### Add Command Feature
+#### Current Implementation
+The sequence diagram below shows the interactions of different subcomponents of the system when adding an expenditure
+to the list.  
+![add_command_sequence_diagram](images/Add_Command_Sequence_Diagram.png)  
```

Comment:
Perhaps there should be an activation bar on the call to the AddCommandInputTest#test(...) method as shown in the screenshot?:
![image](https://user-images.githubusercontent.com/69451253/160956872-c75c4fa6-5c73-44f2-964f-0e1b611570e8.png)

Reply:
the return values should include the data type e.g. category: String

Reply:
This applies to all the return value

Reply:
Would it be possible to enter the objectName:Datatype format for the return types?